qmulticast/utils/multipartite_network.py

```diff
@@ -205,3 +139,67 @@ def create_multipartite_network(name: str, graph: DiGraph) -> Network:
             mem_position += 2
 
     return network
+
+
+def add_mulitpartite_components(node,node_name,port_size,source_delay,source_noise,state_sampler):
+    logger.debug(f"Adding QSource for node 'qsource-{node_name}'.")
+
+    qsource = QSource(
+        name=f"qsource-{node_name}",
+        state_sampler=state_sampler,
```

Comment:
`state_sampler` isn't used anywhere else, it only needs to be variable if you're planning to use different ones.

Reply:
state_sampler depends on the size of the graph, it was either pass this or pass the graph object (or use port size which I thought might be confusing)